### DISCUSSION TITLE
Hello there

### DISCUSSION THREAD

**Discussion by RandomEC** [evennia/evennia#2622] [Greetings & Introductions]
I played muds extensively (mainly Castle Arcanum) in the late 90s, and loved it. Just found out that CA went dark recently, and was inspired to try to recreate at least somewhat of the experience I'd had there, at least for myself and my friends. I had learned the basics of C in fooling around with CA back in the day, but had heard a number of people talking about Evennia, and Python seemed much more accessible than C, even though I'd be starting over. So here I am. ;)

I'm building a mud that is largely going to be hack & slash, with a thematic emphasis on both traditional high fantasy areas and the very-different pop-culture-themed areas (books, movies, music, etc.). I know it's a big turn-off for many people, but I absolutely loved the ability on Castle Arcanum to have a completely anachronistic experience like fighting an elven lord, and then heading off to Pepperland to do battle with Eleanor Rigby or the Walrus. Thus far, I'm loving Python. There's definitely a learning curve, but I'm having fun. I'm sure you'll be hearing questions from me soon, though I've figured out a few things just from searching on here!

Random

**Comment by Aesran** [comment]
Welcome here ! Good luck for your project and you're welcome on discord where I am myself. I think you can look on some project (the others will guide u better) who have maybe a better base than the evennia alone for building Hack'n'Slash. I mean you can start from zero, but it's alway good to take a look on what's already exist.

**Reply by kovitikus** [reply]
Discord Invite Link: https://discord.gg/AJJpcRUhtF

**Reply by RandomEC** [reply]
Thanks! I signed up on Discord the other day, I just haven't posted anything there yet. I was originally intending to steal snippets of things here and there, and I may still, at least for ideas, but I really like the idea of building from the ground up so that you know what you have and  how it works. We'll see how well that survives!

**Comment by kovitikus** [comment]
Glad to have you around!

My main suggestion is to get familiar with the [API](https://www.evennia.com/docs/latest/Evennia-API.html).

Specifically you should become very intimate with the [evennia.objects.objects module](https://www.evennia.com/docs/latest/api/evennia.objects.objects.html#evennia.objects.objects.DefaultObject). It houses all of the most important and useful DefaultObject methods, like `msg(), move_to()`, and `search()`. As well as all of the important properties and hooks!

**Reply by RandomEC** [reply]
This is definitely where I have started. My first step was building a bunch of character attributes and doing character creation, which led me to monkeying around with the Character class, and the CharCreate command. It was a real struggle at first, but got easier. I have just started on creating a system for wearing and removing equipment. All of the ins and outs of equipment is my first real project, I'm going to jump on combat after that. I'm finding making "wear", "remove" and "equipment" commands much easier than working with CharCreate. It's been fun so far!

Edit: Oops! I should also note that your suggestion on the API is spot-on. It is soooooo much easier to find stuff in Evennia than, say, in Diku/Merc/Envy. Even if you don't want to chase everything back in the manner Griatch suggests, everything is so intuitively named that it's fairly easy to find stuff once you have a clue what you are doing.

**Reply by Tegiminis** [reply]
Combat is a good first module to start on, since it often informs the entire rest of your game

**Comment by pruuk** [comment]
I'm also doing a twitch style hack and slash-er. Here is what my combat looked like (on a previous version of the mud):
https://user-images.githubusercontent.com/72707866/117393065-8b54ce00-aeb0-11eb-8439-79dc82bc7263.png

I found the traits contrib and some of the equipment stuff in the Ainneve code really helpful to crib ideas from. It might be worth a look.

**Reply by kovitikus** [reply]
My game is also hack and slash. You and @RandomEC may benefit from poking around my code. Take a gander at my CombatHandler over here.

https://github.com/kovitikus/hecate/tree/master/skills

**Reply by pruuk** [reply]
It's very clean, and I like the body part groupings for targeting (I may incorporate that idea), but it's a bit different than what I'm going for. Also, I have a lot of description text already built for the other system (and I call a different handler to describe resolved actions). I'll link the code once I get combat up and going again. For now, I've been building out the builder menus so my wife will stay enthused about world building.

Very cool setup. Thank you for showing what you're working on!

**Reply by Xinefus-Braska** [reply]
hello @pruuk I am looking at your combat handler/controller and was wondering if you had some time to explain exactly how it works? I am looking for a similar hack and slash style of combat wherein the players/npcs will do 'some' automatic combat, but also require input from the user. 

It looks like you were preparing it to be ready for group combat with range and ability to add/remove players/npcs. It looks pretty interesting and I'd like to learn a bit more about it if possible and perhaps even use parts of it. 

I'm new to py, coming from DIKU/SMAUG C/C++ style muds. Evennia is so neat and I'm enjoying learning as I go. 

Thanks!

**Reply by Xinefus-Braska** [reply]
> My game is also hack and slash. You and @RandomEC may benefit from poking around my code. Take a gander at my CombatHandler over here.
> 
> https://github.com/kovitikus/hecate/tree/master/skills

@kovitikus Do you still have this repository or have you deleted/privated it? I was hoping to find more examples of hack and slash/twitch combat with Evennia. 

Cheers,

**Reply by pruuk** [reply]
@Xinefus-Braska I'd love to sit down (over VC or phone or via email) and talk over how my combat control system works. I'm retiring again tomorrow and one of my projects will be to restart my effort to build my MUD. I'd love to have someone to bounce some ideas off of. If you're willing, reach out to calkdavis@gmail.com

**Comment by Griatch** [comment]
@RandomEC Belated, but welcome! :grinning:

**Reply by RandomEC** [reply]
Thanks! And thanks for all the help!
